knexfile.js

```diff
@@ -1,28 +1,36 @@
+const { knexSnakeCaseMappers } = require('objection');
+
 try {
   // eslint-disable-next-line implicit-dependencies/no-implicit
   require('dotenv').config();
 } catch (e) {}
 
-module.exports = {
-
-  test: {
-    client: 'postgresql',
-    connection: {
-      host: process.env.DATABASE_HOST || 'localhost',
-      database: process.env.DATABASE_NAME,
-      user: process.env.DATABASE_USERNAME || 'asl-test'
-    }
-  },
+let test = {
+  client: 'postgresql',
+  connection: {
+    host: process.env.DATABASE_HOST || 'localhost',
+    database: process.env.DATABASE_NAME,
+    user: process.env.DATABASE_USERNAME || 'asl-test'
+  }
+};
 
-  development: {
-    client: 'postgresql',
-    connection: {
-      host: process.env.DATABASE_HOST,
-      database: process.env.DATABASE_NAME,
-      port: process.env.DATABASE_PORT,
-      user: process.env.DATABASE_USERNAME || 'postgres',
-      password: process.env.DATABASE_PASSWORD
-    }
+let development = {
+  client: 'postgresql',
+  connection: {
+    host: process.env.DATABASE_HOST,
+    database: process.env.DATABASE_NAME,
+    port: process.env.DATABASE_PORT,
+    user: process.env.DATABASE_USERNAME || 'postgres',
+    password: process.env.DATABASE_PASSWORD
   }
+};
 
+if (process.env.SNAKE_MAPPER) {
+  test = { ...test, ...knexSnakeCaseMappers() };
+  development = { ...development, ...knexSnakeCaseMappers() };
+}
```

Comment:
Won't this result in the case mappers always being applied? I was thinking more something like:

```js
const mappers = process.env.MIGRATION ? {} : knexSnakeCaseMappers();

module.exports = {
  test: {
    ...mappers,
    [settings]
  },
  development: {
     ...mappers,
    [settings]
  }
}
```

Reply:
actually, yours does exactly the same thing in a slightly more verbose way. I didn't see the last 2 lines 🤦‍♂️

Reply:
Ah... well in this case you wouldn't set `SNAKE_MAPPER` env var when running the migration command, but I've added it to the npm script for seed?

Reply:
But I can use your syntax to tidy it up.